File: Projects/DMDPDF/PDFBox/pdmodel/documentinterchange/logicalstructure/Revisions.vb
Imports System.Text

Imports FinSeA

Namespace org.apache.pdfbox.pdmodel.documentinterchange.logicalstructure

    '/**
    ' * 
    ' * @author Koch
    ' * @version $Revision: $
    ' *
    ' * @param <T> the type of object to store the revision numbers with
    ' */
    Public Class Revisions(Of T)

        Private objects As List(Of T)
        Private revisionNumbers As List(Of NInteger)

        Private Function getObjects() As List(Of T)
            If (Me.objects Is Nothing) Then
                Me.objects = New ArrayList(Of T)
            End If
            Return Me.objects
        End Function

        Private Function getRevisionNumbers() As List(Of NInteger)
            If (Me.revisionNumbers Is Nothing) Then
                Me.revisionNumbers = New ArrayList(Of NInteger)
            End If
            Return Me.revisionNumbers
        End Function


        '/**
        ' * 
        ' */
        Public Sub New()
        End Sub


        '/**
        ' * Returns the object at the specified position.
        ' * 
        ' * @param index the position
        ' * @return the object
        ' * @throws IndexOutOfBoundsException if the index is out of range
        ' */
        Public Function getObject(ByVal index As Integer) As T ' throws IndexOutOfBoundsException
            Return Me.getObjects().get(index)
        End Function

        '/**
        ' * Returns the revision number at the specified position.
        ' * 
        ' * @param index the position
        ' * @return the revision number
        ' * @throws IndexOutOfBoundsException if the index is out of range
        ' */
        Public Function getRevisionNumber(ByVal index As Integer) As Integer ' throws IndexOutOfBoundsException
            Return Me.getRevisionNumbers().get(index)
        End Function

        '/**
        ' * Adds an object with a specified revision number.
        ' * 
        ' * @param object the object
        ' * @param revisionNumber the revision number
        ' */
        Public Sub addObject(ByVal [object] As T, ByVal revisionNumber As Integer)
            Me.getObjects().add([object])
            Me.getRevisionNumbers().add(revisionNumber)
        End Sub

        '/**
        ' * Sets the revision number of a specified object.
        ' * 
        ' * @param object the object
        ' * @param revisionNumber the revision number
        ' */
        Protected Friend Sub setRevisionNumber(ByVal [object] As T, ByVal revisionNumber As Integer)
            Dim index As Integer = Me.getObjects().indexOf([object])
            If (index > -1) Then
                Me.getRevisionNumbers().set(index, revisionNumber)
            End If
        End Sub

        '/**
        ' * Returns the size.
        ' * 
        ' * @return the size
        ' */
        Public Function size() As Integer
            Return Me.getObjects().size()
        End Function

        Public Overrides Function toString() As String
            Dim sb As New StringBuilder()
            For i As Integer = 0 To Me.getObjects().size()
                If (i > 0) Then sb.Append("; ")
                sb.Append("object=").Append(Me.getObjects().get(i)).Append(", revisionNumber=").Append(Me.getRevisionNumber(i))
            Next
            Return sb.ToString()
        End Function


    End Class

End Namespace

File: Projects/DMDPDF/PDFBox/pdmodel/documentinterchange/logicalstructure/Revisions.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.apache.pdfbox.pdmodel.documentinterchange.logicalstructure;

import java.util.ArrayList;
import java.util.List;

/**
 * 
 * @author Koch
 * @version $Revision: $
 *
 * @param <T> the type of object to store the revision numbers with
 */
public class Revisions<T>
{

    private List<T> objects;
    private List<Integer> revisionNumbers;

    private List<T> getObjects()
    {
        if (Me.objects Is Nothing)
        {
            Me.objects = new ArrayList<T>();
        }
        return Me.objects;
    }

    private List<Integer> getRevisionNumbers()
    {
        if (Me.revisionNumbers Is Nothing)
        {
            Me.revisionNumbers = new ArrayList<Integer>();
        }
        return Me.revisionNumbers;
    }


    /**
     * 
     */
    public Revisions()
    {
    }


    /**
     * Returns the object at the specified position.
     * 
     * @param index the position
     * @return the object
     * @throws IndexOutOfBoundsException if the index is out of range
     */
    public T getObject(int index) throws IndexOutOfBoundsException
    {
        return Me.getObjects().get(index);
    }

    /**
     * Returns the revision number at the specified position.
     * 
     * @param index the position
     * @return the revision number
     * @throws IndexOutOfBoundsException if the index is out of range
     */
    public int getRevisionNumber(int index) throws IndexOutOfBoundsException
    {
        return Me.getRevisionNumbers().get(index);
    }

    /**
     * Adds an object with a specified revision number.
     * 
     * @param object the object
     * @param revisionNumber the revision number
     */
    public void addObject(T object, int revisionNumber)
    {
        Me.getObjects().add(object);
        Me.getRevisionNumbers().add(revisionNumber);
    }

    /**
     * Sets the revision number of a specified object.
     * 
     * @param object the object
     * @param revisionNumber the revision number
     */
    protected void setRevisionNumber(T object, int revisionNumber)
    {
        int index = Me.getObjects().indexOf(object);
        if (index > -1)
        {
            Me.getRevisionNumbers().set(index, revisionNumber);
        }
    }

    /**
     * Returns the size.
     * 
     * @return the size
     */
    public int size()
    {
        return Me.getObjects().size();
    }

    /**
     * {@inheritDoc}
     */
    public String toString()
    {
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < Me.getObjects().size(); i++)
        {
            if (i > 0)
            {
                sb.append("; ");
            }
            sb.append("object=").append(Me.getObjects().get(i))
                .append(", revisionNumber=").append(Me.getRevisionNumber(i));
        }
        return sb.toString();
    }

}
